SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -199,6 +201,15 @@ void CFluidIteration::Update(COutput* output, CIntegration**** integration, CGeo
                                                                        config[val_iZone], MESH_0);
       integration[val_iZone][val_iInst][TRANS_SOL]->SetConvergence(false);
     }
+
+    /*--- Update dual time solver for the weakly coupled energy equation ---*/
+
+    if (config[val_iZone]->GetWeakly_Coupled_Heat()) {
+      integration[val_iZone][val_iInst][HEAT_SOL]->SetDualTime_Solver(geometry[val_iZone][val_iInst][MESH_0],
+                                                                      solver[val_iZone][val_iInst][MESH_0][HEAT_SOL],
+                                                                      config[val_iZone], MESH_0);
+      integration[val_iZone][val_iInst][HEAT_SOL]->SetConvergence(false);
+    }
```

Comment:
@TobiKattmann @oleburghardt , I think this was missing for dual time weakly coupled stuff, can you see if it makes sense?

Reply:
LGTM. Although I didn't use the `WEAKLY_COUPLED_HEAT` in a long time tbh and for now do not intend to use in the future. 
I also checked: There is a Testcase `discadj_heat` using `WEAKLY_COUPLED_HEAT` (although it is of course not unsteady)

Thanks for actualy thinking of that 👍

Reply:
Alright, thanks for having a look. I don't blame you, there is not a lot going for weakly coupled if the flow solver still drags temperature around. Otherwise having 4 variables (in 3D) would make computers very happy.

Reply:
Ok i dont get the last sentence :) Do you mean the unnecessary energy-equation that is carried around even for cases without it being active.  Or did I miss the memo that states `4>>5` for #equations. Please elaborate if you have a few moments to spare ✍️

Reply:
Yes that too since it makes the linear system 1.44 times larger and slower (cost of sparse operations is proportional to amount of memory used).
Then 4 is a really good number because all the blocks in the matrix would be aligned and all the operations would pack perfectly into AVX registers.
The vector blocks would also be aligned which would make prefetching (loading into cache ahead of time) effective because it operates on cache lines (and 8 doubles in one of those).
Finally the output vector could be written to with non-temporal stores (algo because of alignment) which improves caching of the input vector.
So in all these ways 4 > 5 (by about 5-10% maybe on top of the 1.44).

Reply:
Oh yes sure, if forgot that computers had this thing going with powers of 2 🤦 thanks!